### ISSUE TITLE
Is this documentation outdated?

### ISSUE THREAD

**Issue by joshuap** [honeybadger-io/honeybadger-elixir#301]
The README and code docs for `Honeybadger.notify` say to pass the context as the second argument:

https://github.com/honeybadger-io/honeybadger-elixir/blob/d051f0754dd44e13c9cc7c989d7fdfc5cb068f82/lib/honeybadger.ex#L44

...but it looks like the second argument is actually metadata, which expects a `context` key:

https://github.com/honeybadger-io/honeybadger-elixir/blob/c2dc31100fb764ad4b0c4df6bf94bdef0bbc8984/lib/honeybadger/notice.ex#L58

Should it be `Honeybadger.notify(exception, context)`, or `Honeybadger.notify(exception, %{context: context})`? I may be forgetting how this works.

cc @rabidpraxis @sorentwo

**Comment by sorentwo** [comment]
@joshuap The second argument is the metadata and not the context. To pass in context you need the `%{context: context}` map.

**Comment by rabidpraxis** [comment]
@joshuap `Notice.new` expects a map with `:context`, but it looks like `Honeybadger.notify` will jam any metadata it does not recognize into a context key:

https://github.com/honeybadger-io/honeybadger-elixir/blob/24edb30cb2a2c9462bdc453c9943fa64ef9bac00/lib/honeybadger.ex#L253-L257

https://github.com/honeybadger-io/honeybadger-elixir/blob/24edb30cb2a2c9462bdc453c9943fa64ef9bac00/lib/honeybadger.ex#L428-L435

**Comment by joshuap** [comment]
@rabidpraxis oh OK, so our docs are actually valid? Is there a method you prefer?

**Comment by rabidpraxis** [comment]
I'm not sure. I probably swing towards favoring the explicit version, just for readability sake.

**Comment by hypernumbers** [comment]
The README is - it tells you to go and watch an Elixir Sips screencast - and the link doesn't work

https://hexdocs.pm/honeybadger/readme.html

**Comment by gordonguthrie** [comment]
> The README is - it tells you to go and watch an Elixir Sips screencast - and the link doesn't work
> 
> https://hexdocs.pm/honeybadger/readme.html

This is corrected in the github repo - so please ignore

**Comment by subzero10** [comment]
Yeah, the markdown is not rendered as expected in hexdocs. You'll be redirected to the screencast if you click on the html tag that _appears_ to be an image:
![CleanShot 2024-11-30 at 12 35 40@2x](https://github.com/user-attachments/assets/b991dbfd-c037-42c7-9773-74cc5686fdfa)

But if you click on "Watch our screencast", which is right below, you'll be redirected to a login page. I will correct that to take you to the screencast.
